### PR TITLE
Revert removal of default value for form method

### PR DESCRIPTION
**Pull request by dvic**

### CODE REVIEW

**Comment by dvic** [comment]
In my repo I get the following error for dead views:

```
     ** (FunctionClauseError) no function clause matching in Phoenix.LiveView.Helpers.form_method/1

     The following arguments were given to Phoenix.LiveView.Helpers.form_method/1:
     
         # 1
         nil
     
     Attempted function clauses (showing 2 out of 2):
     
         defp form_method(method) when method === "get" or method === "post"
         defp form_method(method) when is_binary(method)
     
     code: post(conn, Routes.user_session_path(conn, :create, "en"), %{
     stacktrace:
       (phoenix_live_view 0.18.0-dev) lib/phoenix_live_view/helpers.ex:1158: Phoenix.LiveView.Helpers.form_method/1
       (phoenix_live_view 0.18.0-dev) lib/phoenix_live_view/helpers.ex:1117: Phoenix.LiveView.Helpers."form (overridable 1)"/1
       (phoenix_live_view 0.18.0-dev) lib/phoenix_live_view/helpers.ex:619: Phoenix.LiveView.Helpers.component/2
       (sharefox 0.1.0) lib/sharefox_web/templates/user_session/new.html.heex:3: anonymous fn/2 in SharefoxWeb.UserSessionView."new.html"/1
       (phoenix_live_view 0.18.0-dev) lib/phoenix_live_view/engine.ex:124: Phoenix.HTML.Safe.Phoenix.LiveView.Rendered.to_iodata/1
       (phoenix_live_view 0.18.0-dev) lib/phoenix_live_view/engine.ex:140: Phoenix.HTML.Safe.Phoenix.LiveView.Rendered.to_iodata/3
       (phoenix 1.6.9) lib/phoenix/controller.ex:772: Phoenix.Controller.render_and_send/4
```

I tried adding a test for this but I can't replicate this error (tried using the `Phoenix.View.render(View, ..)` tests.

File: lib/phoenix_live_view/helpers.ex [hunk 1]
```diff
@@ -1113,7 +1113,7 @@ defmodule Phoenix.LiveView.Helpers do
     # unless the action is given.
     {attrs, hidden_method, csrf_token} =
       if action do
-        {method, opts} = Keyword.pop(opts, :method)
+        {method, opts} = Keyword.pop(opts, :method, "post")
```
Comment (feliperenan) [inline lib/phoenix_live_view/helpers.ex:1116]: One way to reproduce this is creating a test with `action="/"` without a http method in here:

https://github.com/phoenixframework/phoenix_live_view/blob/master/test/phoenix_live_view/helpers_test.exs#L90

Something like this:

```eex
      html =
        parse(~H"""
        <.form :let={f} for={:myform} action="/">
          <%= text_input f, :foo %>
        </.form>
        """)
```

Would raise the error you are facing so we can cover this up :)

Comment (dvic) [reply lib/phoenix_live_view/helpers.ex:1116]: > One way to reproduce this is creating a test with `action="/"` without a http method in here:
> 
> https://github.com/phoenixframework/phoenix_live_view/blob/master/test/phoenix_live_view/helpers_test.exs#L90
> 
> Something like this:
> 
> ```
>       html =
>         parse(~H"""
>         <.form :let={f} for={:myform} action="/">
>           <%= text_input f, :foo %>
>         </.form>
>         """)
> ```
> 
> Would raise the error you are facing so we can cover this up :)

@feliperenan Indeed, that's the test that triggers the error :) Thanks!

File: test/phoenix_live_view/helpers_test.exs [hunk 2]
```diff
@@ -100,6 +100,18 @@ defmodule Phoenix.LiveView.HelpersTest do
...
+    test "does not raise when missing method" do
+      assigns = %{}
+
+      html =
+        parse(~H"""
+        <.form for={:myform} action="/">
+        </.form>
+        """)
+
+        assert [{"form", [{"action", "/"}, {"method", "post"}], _}] = html
```
Comment (feliperenan) [inline test/phoenix_live_view/helpers_test.exs:112]:
```suggestion
      assert [{"form", [{"action", "/"}, {"method", "post"}], _}] = html
```

**Review by feliperenan — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**